Add Virtual Quota Provider Usage Tracking & Race Condition Check

mcowger opened the pull request:
## Context

- **Add progress bars to help user understand current limit progress.**
- **Cooldown period enforcement and race confition fix.**
- **Add changeset**


## Implementation

* Stronger checks on active Handler selection
* webView messages pass consumption data back and forth.

## How to Test

* Configure virtual quota provider.
* Perform some operatations
* See that progress bars are filling.

## Get in Touch

mcowger on Discord

mcowgeratsigma committed:
Add progress bars to help user understand current limit progress.

mcowgeratsigma committed:
Cooldown period enforcement and race confition fix.

feat: enhance VirtualQuotaFallbackHandler with cooldown management and usage tracking
bugfix: Address a race condition during startup with OpenRouter-based providers (Kilocode and Openrouter) where using before they have fetched their models leads to aborted tasks during image-cleaning.

mcowgeratsigma committed:
Add changeset

mcowgeratsigma committed:
Address mocking/race condition in the tests.

chrarnoldus commented on .changeset/yellow-years-fry.md:
```diff
@@ -0,0 +1,6 @@
+---
+"kilo-code": minor
+"@roo-code/types": patch
```
```suggestion
```

hassoncs commented:
Thanks for opening this PR, @mcowger! We fixed the playwright tests so if you rebase against `main` the build should be green! I'll review it shortly

mcowger commented:
@hassoncs - rebased and still same issue with playwright

Commit:
fix(virtual-quota): Correct usage tracking and cooldown logic

Refactor VirtualQuotaFallbackHandler to correctly track usage and apply cooldowns.
Changed `AllUsageResult` to `UsageResultByDuration` for clarity.
Ensured `updateProfileData` message is sent to webview on error.

Commit:
feat(ui): add LabeledProgress component and integrate into VirtualQuotaFallbackProvider

hassoncs commented:
@mcowger sorry for the delay on this! I'm wrapping up the review now- awesome work. I'm moving the progress bars to under the input boxes, (one bar per box to avoid needing separate labels) but other than that looks solid!

Commit:
refactor(ui): extract VirtualQuotaFallbackProviderPresentation

Extracts the presentation logic for `VirtualQuotaFallbackProvider` into a new `VirtualQuotaFallbackProviderPresentation` component. This separates the UI rendering from the component's state management and logic, improving maintainability and testability.

The `VirtualQuotaFallbackProvider` now handles state, data fetching, and event handlers, passing necessary props to the new presentation component. The `VirtualLimitInputs` component has also been renamed to `VirtualLimitInputsPresentation` and moved into the new presentation file.

This change also updates the Storybook story to use the new presentation component and adjusts the `LabeledProgress` component's styling for better visual integration.

Commit:
ci: update CI workflow condition

Refactor the condition for running Playwright E2E tests in the Storybook snapshot workflow to explicitly check for 'true' value of SECRETS_SET.

hassoncs reviewed (approved):
Great stuff! Thanks again, @mcowger !!
